miner/worker_builder.go

```diff
@@ -435,6 +448,15 @@ func (w *worker) simulateBundle(
 		}
 
 		if receipt.Status == types.ReceiptStatusFailed && !containsHash(bundle.RevertingTxHashes, receipt.TxHash) {
+			// for unRevertible tx but itself can be dropped, we drop it and revert the state and gas pool
+			if containsHash(bundle.DroppingTxHashes, receipt.TxHash) {
+				log.Warn("drop tx in bundle", "hash", receipt.TxHash.String())
+				state = prevState
+				gasPool = prevGasPool
+				bundle.Txs = bundle.Txs.Remove(i)
```

Comment:
if len bundle.Txs ==0, we should drop the bundle

Reply:
it has been checked on line500, after finish the txs loop